### PR TITLE
Parsenodes fix: check if index is out-of-bound first

### PR DESCRIPTION
**Pull request by merkispavel**
# Issue

if the code above results in `curren_way_node_index_ == npos` then  vector out-of-range error happens
```cpp
if (osmid > (*(*way_nodes_)[current_way_node_index_]).node.osmid_) {
      current_way_node_index_ =
          way_nodes_->find_first_of(OSMWayNode{{osmid}},
                                    [](const OSMWayNode& a, const OSMWayNode& b) {
                                      return a.node.osmid_ <= b.node.osmid_;
                                    },
                                    current_way_node_index_);
    }
```

## Tasklist

 - [ ] Add tests
 - [ ] Add #fixes with the issue number that this PR addresses
 - [ ] Update the docs with any new request parameters or changes to behavior described
 - [ ] Update the [changelog](CHANGELOG.md)

## Requirements / Relations

 Link any requirements here. Other pull requests this PR is based on?

### CODE REVIEW

File: src/mjolnir/pbfgraphparser.cc [hunk 1]
```diff
@@ -855,8 +855,8 @@ struct graph_callback : public OSMPBF::Callback {
     // if this nodes id is less than the waynode we are looking for then we know its a node we can
     // skip because it means there were no ways that we kept that referenced it. also we could run out
     // of waynodes to look for and in that case we are done as well
-    if (osmid < (*(*way_nodes_)[current_way_node_index_]).node.osmid_ ||
-        current_way_node_index_ >= way_nodes_->size()) {
+    if (current_way_node_index_ >= way_nodes_->size() ||
```
Comment (danpat) [inline src/mjolnir/pbfgraphparser.cc:858]: Should this be `>`, not `>=` ?  If `current_way_node_index_` == `way_nodes_->size()`, then we'll access 1 past the end of the array, even with your fix.

Comment (merkispavel) [reply src/mjolnir/pbfgraphparser.cc:858]: if `current_way_node_index_  == way_nodes_->size()` then it immediately jump to `return;` without checking the second condition, nope?

Comment (danpat) [reply src/mjolnir/pbfgraphparser.cc:858]: 🤦 I had it backwards, nevermind!

File: src/mjolnir/pbfgraphparser.cc [hunk 2]
```diff
@@ -855,8 +855,8 @@ struct graph_callback : public OSMPBF::Callback {
     // if this nodes id is less than the waynode we are looking for then we know its a node we can
     // skip because it means there were no ways that we kept that referenced it. also we could run out
     // of waynodes to look for and in that case we are done as well
-    if (osmid < (*(*way_nodes_)[current_way_node_index_]).node.osmid_ ||
-        current_way_node_index_ >= way_nodes_->size()) {
```
Comment (kevinkreiser) [inline src/mjolnir/pbfgraphparser.cc:859]: derp, good catch, sorry about that!

**Review by kevinkreiser — APPROVED** [review]

**Review by gknisely — DISMISSED** [review]

**Pull request merged**